### PR TITLE
Add GHANA NLP TTS API v1

### PR DESCRIPTION
**Pull request by anish-work**
### Q/A checklist

- [x] If you add new dependencies, did you update the lock file?
```bash
poetry lock --no-update
```
- [x] Run tests 
```bash
ulimit -n unlimited && ./scripts/run-tests.sh
```
- [x] Do a self code review of the changes - Read the diff at least twice.  
- [x] Carefully think about the stuff that might break because of this change - this sounds obvious but it's easy to forget to do "Go to references" on each function you're changing and see if it's used in a way you didn't expect. 
- [x] The relevant pages still run when you press submit
- [x] The API for those pages still work (API tab)
- [x] The public API interface doesn't change if you didn't want it to (check API tab > docs page)
- [x] Do your UI changes (if applicable) look acceptable on mobile?
- [x] Ensure you have not regressed the import time unless you have a good reason to do so. 
You can visualize this using tuna:
```bash
python3 -X importtime -c 'import server' 2> out.log && tuna out.log
```
To measure import time for a specific library:
```bash
$ time python -c 'import pandas'

________________________________________________________
Executed in    1.15 secs    fish           external
   usr time    2.22 secs   86.00 micros    2.22 secs
   sys time    0.72 secs  613.00 micros    0.72 secs
```
To reduce import times, import libraries that take a long time inside the functions that use them instead of at the top of the file:
```python
def my_function():
    import pandas as pd
    ...
```

### Legal Boilerplate

Look, I get it. The entity doing business as “Gooey.AI” and/or “Dara.network” was incorporated in the State of Delaware in 2020 as Dara Network Inc. and is gonna need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Dara Network Inc can use, modify, copy, and redistribute my contributions, under its choice of terms.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205803182950231

### CODE REVIEW

File: recipes/TextToSpeech.py [hunk 1]
```diff
@@ -401,6 +402,22 @@ def run(self, state: dict):
                 state["audio_url"] = upload_file_from_bytes(
                     "openai_tts.mp3", response.content
                 )
+            case TextToSpeechProviders.GHANA_NLP:
+                data = requests.post(
```
[the remark this answers is not in the document]
Comment (anish-work) [reply recipes/TextToSpeech.py]: @devxpy updated

File: recipes/TextToSpeech.py [hunk 2]
```diff
@@ -401,6 +403,18 @@ def run(self, state: dict):
                 state["audio_url"] = upload_file_from_bytes(
                     "openai_tts.mp3", response.content
                 )
+            case TextToSpeechProviders.GHANA_NLP:
+                response = requests.post(
+                    "https://translation-api.ghananlp.org/tts/v1/tts",
+                    headers=GHANA_API_AUTH_HEADERS,
+                    json={
+                        "text": text,
+                        "language": "tw",
```
Comment (devxpy) [inline recipes/TextToSpeech.py:412]: Shouldnt this use the language that the user selected

File: daras_ai_v2/asr.py [hunk 3]
```diff
@@ -202,7 +202,7 @@
 GHANA_NLP_SUPPORTED = {'en': 'English', 'tw': 'Twi', 'gaa': 'Ga', 'ee': 'Ewe', 'fat': 'Fante', 'dag': 'Dagbani',
                        'gur': 'Gurene', 'yo': 'Yoruba', 'ki': 'Kikuyu', 'luo': 'Luo', 'mer': 'Kimeru'}  # fmt: skip
 GHANA_NLP_MAXLEN = 500
-
+GHANA_API_AUTH_HEADERS = {"Ocp-Apim-Subscription-Key": str(settings.GHANA_NLP_SUBKEY)}
```
Comment (devxpy) [inline daras_ai_v2/asr.py:205]: This same constant should also be used in asr.py

File: recipes/TextToSpeech.py [hunk 4]
```diff
@@ -57,7 +59,7 @@ class TextToSpeechSettings(BaseModel):
 
     openai_voice_name: OpenAI_TTS_Voices.api_choices | None
     openai_tts_model: OpenAI_TTS_Models.api_choices | None
-
+    ghana_tts_language: GHANA_TTS_LANGUAGES.api_choices | None
```
Comment (devxpy) [inline recipes/TextToSpeech.py:62]: Please change to `ghana_nlp_tts_language` given "ghana tts" doesnt show the right thing when you google vs "ghana nlp tts" and also for consistency with the rest of the codebase that uses `ghana_nlp_*` prefix

File: daras_ai_v2/asr.py (hunk 3 above)
Comment (anish-work) [reply daras_ai_v2/asr.py:205]: @devxpy I added that in the other PRs too so I thought let's not do the same here

File: recipes/TextToSpeech.py (hunk 2 above)
Comment (anish-work) [reply recipes/TextToSpeech.py]: @devxpy updated

Comment (devxpy) [reply recipes/TextToSpeech.py]: This should read from local var state not global gui.session state

File: recipes/TextToSpeech.py (hunk 4 above)
Comment (anish-work) [reply recipes/TextToSpeech.py]: @devxpy updated

File: recipes/TextToSpeech.py (hunk 2 above)
Comment (anish-work) [reply recipes/TextToSpeech.py]: @devxpy updated used the state var but I noticed that open is using gui.session_state so I used the same

**Comment by devxpy** [comment]
lgtm, but please make sure you test localhost:3000/lipsynctts and localhost:3000/copilot on ghana nlp tts

**Pull request merged**